client/image_tag.go

```diff
@@ -12,10 +12,15 @@ import (
 )
 
 // ImageTag tags an image in the docker host
-func (cli *Client) ImageTag(ctx context.Context, imageID, ref string) error {
-	distributionRef, err := distreference.ParseNamed(ref)
+func (cli *Client) ImageTag(ctx context.Context, source, target string) error {
+	_, err := distreference.ParseNamed(source)
 	if err != nil {
-		return fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", ref)
+		return fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", source)
```

Comment:
I realize the original checks don't do this either, but perhaps we should use the original error, as it may contain useful information. Something like;

```patch
diff --git a/client/image_tag.go b/client/image_tag.go
index 07b92a6..dbcd078 100644
--- a/client/image_tag.go
+++ b/client/image_tag.go
@@ -1,26 +1,23 @@
 package client

 import (
-       "errors"
-       "fmt"
        "net/url"

-       "golang.org/x/net/context"
-
        distreference "github.com/docker/distribution/reference"
        "github.com/docker/docker/api/types/reference"
+       "github.com/pkg/errors"
+       "golang.org/x/net/context"
 )

 // ImageTag tags an image in the docker host
 func (cli *Client) ImageTag(ctx context.Context, source, target string) error {
-       _, err := distreference.ParseNamed(source)
-       if err != nil {
-               return fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", source)
+       if _, err := distreference.ParseNamed(source); err != nil {
+               return errors.Wrapf(err, "Error parsing reference: %q is not a valid repository/tag", source)
        }

        distributionRef, err := distreference.ParseNamed(target)
        if err != nil {
-               return fmt.Errorf("Error parsing reference: %q is not a valid repository/tag", target)
+               return errors.Wrapf(err, "Error parsing reference: %q is not a valid repository/tag", target)
        }

        if _, isCanonical := distributionRef.(distreference.Canonical); isCanonical {
```

Which produces;

```
$ docker tag HELLO-WORLD foobar
Error parsing reference: "HELLO-WORLD" is not a valid repository/tag: repository name must be lowercase

$ docker tag hello-world:latest@FOOBAR foobar
Error parsing reference: "hello-world:latest@FOOBAR" is not a valid repository/tag: invalid reference format
```